src/sqllogictest/src/runner.rs

```diff
@@ -580,13 +580,15 @@ impl Runner {
                 tracing: TracingCliArgs::default(),
             },
             secrets_controller: None,
-            listen_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
+            sql_listen_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
+            http_listen_addr: SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0),
```

Comment:
will this not fail to setup the second listener if the ports are the same?

Reply:
I originally wondered this too, but it turns out that when you set an address to port 0, the os will dynamically assign an available port. https://www.lifewire.com/port-0-in-tcp-and-udp-818145